src/main/scala/uk/gov/homeoffice/drt/repository/ArrivalRepository.scala

```diff
@@ -40,32 +45,32 @@ class ArrivalRepository[F[_] : Sync](val sessionPool: Resource[F, Session[F]]) e
   implicit val logger = Slf4jLogger.getLogger[F]
 
   val decoder: Decoder[ArrivalTableData] =
-    (text ~ int4 ~ text ~ text ~ text ~ text ~ int4.opt ~ timestamp ~ timestamp.opt).map {
-      case code ~ number ~ destination ~ origin ~ terminal ~ status ~ totalpassengers ~ scheduled ~ scheduled_departure  =>
+    (text ~ int4 ~ text ~ text ~ text ~ text ~ int4.opt ~ timestamp ~ timestamp.opt ~ timestamp.opt ~ timestamp.opt ~ timestamp.opt ~ timestamp.opt ~ timestamp.opt).map {
+      case code ~ number ~ destination ~ origin ~ terminal ~ status ~ totalpassengers ~ scheduled ~ estimated ~ actual ~ estimatedchox ~ actualchox ~ pcp ~ scheduled_departure =>
         ArrivalTableData(
-          code, number, destination, origin, terminal, status, totalpassengers, scheduled, scheduled_departure
+          code, number, destination, origin, terminal, status, totalpassengers, scheduled, estimated, actual, estimatedchox, actualchox, pcp, scheduled_departure
         )
     }
 
 
-  private def selectArrivalsForADate: Query[LocalDateTime ~ LocalDateTime, ArrivalTableData] =
+  private def selectArrivalsForOriginsAndADate(originsSize: Int): Query[List[String] ~ LocalDateTime ~ LocalDateTime, ArrivalTableData] =
     sql"""
-        SELECT code, number, destination, origin, terminal, status, totalpassengers, scheduled, scheduled_departure
-        FROM arrival WHERE scheduled > $timestamp and scheduled < $timestamp;
+        SELECT code, number, destination, origin, terminal, status, totalpassengers, scheduled, estimated , actual , estimatedchox , actualchox , pcp ,scheduled_departure
+        FROM arrival WHERE origin in(${text.list(originsSize)}) and scheduled > $timestamp and scheduled < $timestamp;
```

Comment:
Just checking the logic here. Surely `scheduled` can't be both greater and less than `$timestamp`?

Reply:
it is type of parameter rather than value it is prepared query . value is passed when ps.stream is called.